### PR TITLE
Add property attribute point clouds sample

### PR DESCRIPTION
**Pull request by javagl**
A **DRAFT** for an example showing point clouds with property attributes and statistics.

### CODE REVIEW

**Comment by javagl** [comment]
Updated the sample to add a README, screenshot and sandcastle. 

Preview: https://github.com/javagl/3d-tiles-samples/tree/add-property-attribute-point-clouds/glTF/EXT_structural_metadata/PropertyAttributesPointCloud 

The line
```
int classification = int(fsInput.metadata.classification + 0.5);
```
is still to be discussed: It should not be necessary to round up there.

**Comment by lilleyse** [comment]
sRGB -> linear conversion needed here?

Selection_031

**Comment by lilleyse** [comment]
> The line
> ```
> int classification = int(fsInput.metadata.classification + 0.5);
> ```
> is still to be discussed: It should not be necessary to round up there.
> 

It seems like the attributes are being interpolated even though the mode is `gl.POINTS`. My theory is that the GPU is treating each point as a quad, duplicating the attributes at each corner, and running interpolation math. I wonder if the `flat` qualifier in WebGL 2 would fix this.

I put together some custom shaders to help debug this.

### Version 1: vertex shader only approach - works!
```js
  CLASSIFICATION_SHADER: new Cesium.CustomShader({
    varyings: {
      v_color: Cesium.VaryingType.VEC3,
    },
    vertexShaderText: `
    void vertexMain(VertexInput vsInput, inout czm_modelVertexOutput vsOutput) {
        int classif = int(vsInput.attributes.classification);
        
        vec3 color = vec3(1);
        if (classif == 0) {
            color = vec3(0,0.5,0);
        }
        else if (classif == 1) {
            color = vec3(0.5,0.5,0.5);
        }

        v_color = color;
    }`,
    fragmentShaderText: `
    void fragmentMain(FragmentInput fsInput, inout czm_modelMaterial material)
    {
        material.diffuse = v_color;
    }
    `,
  }),
```

image

### Version 2: fragment shader approach with custom varying - fails!

```js
  CLASSIFICATION_SHADER: new Cesium.CustomShader({
    varyings: {
      v_classif: Cesium.VaryingType.FLOAT,
    },
    vertexShaderText: `
    void vertexMain(VertexInput vsInput, inout czm_modelVertexOutput vsOutput) {
        v_classif = vsInput.attributes.classification;
    }`,
    fragmentShaderText: `
    void fragmentMain(FragmentInput fsInput, inout czm_modelMaterial material)
    {
        vec3 color = vec3(1);
        int classification = int(v_classif);
        if (classification == 0) {
            color = vec3(0,0.5,0);
        }
        else if (classification == 1) {
            color = vec3(0.5,0.5,0.5);
        }
        material.diffuse = color;
    }
    `,
  }),
```


image

**Comment by javagl** [comment]
I have updated the colors with sRGB (not sure whether it's worth updating the screenshot as well), and applied the workaround that @lilleyse suggested. The issue of imprecise `int` values in the fragment shader is now tracked at https://github.com/CesiumGS/cesium/issues/10699 (also linked in the sandcastle), so I think this is ready for review.

File: glTF/EXT_structural_metadata/PropertyAttributesPointCloud/README.md [hunk 1]
```diff
@@ -69,19 +69,33 @@ const shaders = {
...
   CLASSIFICATION_SHADER: new Cesium.CustomShader({
-    fragmentShaderText: `
-    void fragmentMain(FragmentInput fsInput, inout czm_modelMaterial material)
-    {
+    varyings: {
+      v_color: Cesium.VaryingType.VEC3,
+    },
+    vertexShaderText: `
+    void vertexMain(VertexInput vsInput, inout czm_modelVertexOutput vsOutput) {
+        int classif = int(vsInput.attributes.classification);
```
Comment (lilleyse) [inline glTF/EXT_structural_metadata/PropertyAttributesPointCloud/README.md:83]: I was kind of sloppy when I called this `classif`... you can rename back to `classification`

**Comment by javagl** [comment]
I actually had noticed that in the comment, but didn't update it when copying the shader from the comment to the sandcastle. Updated.

**Pull request merged**